### PR TITLE
Multi queries

### PR DESCRIPTION
**Pull request by pandu-k**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature

* **What is the current behavior?** (You can also link to an open issue here)
No way for weighted multi query searches

* **What is the new behavior (if this is a feature change)?**
Weighted multi query searches. 

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

* **Have unit tests been run against this PR?** (Has there also been any additional testing?)
Yes

* **Related Python client/documentation etc changes** (link commit/PR here)
Client: https://github.com/marqo-ai/py-marqo/pull/54/
API tests: https://github.com/marqo-ai/marqo-api-tests/pull/9
Documentation: https://gh-previews.marqo.pages.dev/0.0.13/API-Reference/search/#query-q

* **Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: src/marqo/s2_inference/s2_inference.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def vectorise(model_name: str, content: Union[str, List[str]], model_properties:
     try:
         vectorised = available_models[model_cache_key].encode(content, normalize=normalize_embeddings, **kwargs)
     except UnidentifiedImageError as e:
-        raise VectoriseError from e
+        raise VectoriseError(str(e)) from e
```
Comment (pandu-k) [inline src/marqo/s2_inference/s2_inference.py:48]: This adds the UnidentifiedImageError message to e

File: src/marqo/tensor_search/tensor_search.py [hunk 2]
```diff
@@ -1182,9 +1214,10 @@ def _vector_text_search(
                     "Try reducing the query's limit parameter") from e
             elif 'parse_exception' in response["responses"][0]["error"]["root_cause"][0]["reason"]:
                 raise errors.InvalidArgError("Syntax error, could not parse filter string") from e
-            elif contextualised_filter in response["responses"][0]["error"]["root_cause"][0]["reason"]:
+            elif (contextualised_filter
+                  and contextualised_filter in response["responses"][0]["error"]["root_cause"][0]["reason"]):
                 raise errors.InvalidArgError("Syntax error, could not parse filter string") from e
-            raise e
+            raise errors.BackendCommunicationError(f"Error communicating with Marqo-OS backend:\n{response}")
```
Comment (pandu-k) [inline src/marqo/tensor_search/tensor_search.py:1220]: We pass through the Marqo-os error as fallback, if not yet handled

File: src/marqo/s2_inference/processing/custom_clip_utils.py [hunk 3]
```diff
@@ -6,7 +6,9 @@
 import os
 import urllib
 from tqdm import tqdm
-from src.marqo.s2_inference.configs import ModelCache
```
Comment (vicilliar) [inline src/marqo/s2_inference/processing/custom_clip_utils.py]: mentioned this bug to li, i think he might have made the same change. looks good.

File: src/marqo/tensor_search/tensor_search.py [hunk 4]
```diff
@@ -1049,12 +1055,38 @@ def _vector_text_search(
         raise errors.IndexNotFoundError(message="Tried to search a non-existent index: {}".format(index_name))
     selected_device = config.indexing_device if device is None else device
 
-    # TODO average over vectorized inputs with weights
+    # query, weight pairs, if query is a dict:
+    ordered_queries = None
+
+    if isinstance(query, str):
+        to_be_vectorised = [query, ]
```
Comment (vicilliar) [inline src/marqo/tensor_search/tensor_search.py:1062]: can this just be `to_be_vectorised = [query]`?

File: src/marqo/tensor_search/tensor_search.py [hunk 5]
```diff
@@ -1049,12 +1055,38 @@ def _vector_text_search(
...
+    else:  # is dict:
+        ordered_queries = list(query.items())
+        if index_info.index_settings[NsField.index_defaults][NsField.treat_urls_and_pointers_as_images]:
+            text_queries = [k for k, _ in ordered_queries if _is_image(k)]
+            image_queries = [k for k, _ in ordered_queries if not _is_image(k)]
```
Comment (vicilliar) [inline src/marqo/tensor_search/tensor_search.py:1067]: could these be extracted in 1 loop to make it faster? like
```
for k, _ in ordered_queries:
   if _is_image(k):
      text_queries.append(k)
   else:
      image_queries.append(k)
```

**Review by vicilliar — APPROVED** [review]

**Pull request merged**